### PR TITLE
Adopt juju 3.1

### PR DESCRIPTION
**Pull request by dashmage**

### CODE REVIEW

**Review by rgildein — APPROVED** [review]
LGTM

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  func:
+    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    needs: lint-unit
+    with:
+      command: make functional
+      snapcraft: true
+      juju-channel: "3.1/stable"
+      nested-containers: true
+      python-version: "3.10"
+      timeout-minutes: 120
```
Comment (rgildein) [inline .github/workflows/check.yaml:32]: realized that this is not needed to be 120min, but it's still non blocking

Comment (dashmage) [reply .github/workflows/check.yaml]: Done! Changed to 60.

**Review by rgildein — APPROVED** [review]

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Check workflow running linter, unit and functional tests
```
Comment (agileshaw) [inline .github/workflows/check.yaml:1]: We need to change `sonar.yaml` ([L#9](https://github.com/canonical/prometheus-juju-backup-all-exporter/blob/main/.github/workflows/sonar.yaml#L9C11-L9C11) and [L#17](https://github.com/canonical/prometheus-juju-backup-all-exporter/blob/main/.github/workflows/sonar.yaml#L17)) to use this workflow name

**Review by agileshaw — CHANGES_REQUESTED** [review]

**Review by agileshaw — APPROVED** [review]
LGTM, thanks

File: .github/workflows/check.yaml (hunk 2 above)
Comment (rgildein) [reply .github/workflows/check.yaml:1]: Och I did no realized that and there are potentially more name like this. @dashmage Can I ask you to check to fix it also in other projects (I think [this](https://github.com/search?q=org%3Acanonical%20%22name%3A%20Check%20workflow%20running%20linter%2C%20unit%20and%20functional%20tests%22&type=code) is full lists), thanks.

**Review by rgildein — APPROVED** [review]

**Pull request merged**